src/models/util/ParseCalChart3File.ts

```diff
@@ -0,0 +1,251 @@
+ * In general, the layout is <4char> <size_of_data> <data...> <"END "> <4char>, with
+ * the exception of the first "INGL" block and the version
+ * the Version block is either "GURK", for CalChart3.0, or "GUXY" where X and Y are version 
+ * numbers in hex.
+ * 
+ * For CalChart 3.3 and later there are consistent 4char blocks that contain more elements
+ * of the show.
+ */
+
+import Show from '../Show';
+import StuntSheet from '../StuntSheet';
+import StuntSheetDot from '../StuntSheetDot';
+
+/**
+ * LoadShow
+ * 
+ * @param buffer The Byte array of data to load.
+ * @returns Returns either a new [Show] or null if the Byte array is ill-formed.
+ */
+export default function LoadShow(buffer : ArrayBuffer) {
+  try {
+    // parsing frequently involves throwing to indicate the parse failed.
+    return LoadShowHelper(buffer);
+  }
+  catch(e) {
+    console.log(e);
+  }
+  return null;
+}
+
+function LoadShowHelper(buffer : ArrayBuffer) {
+  // first we parse and confirm the file is well formed
+  if (!ReadAndCheckID(new DataView(buffer, 0, 4), 0, "INGL")) {
+    return null;
+  }
+  // Extract the version number
+  const version = ReadVersion(new DataView(buffer, 4, 4));
+  if (version <= 33) {
+    console.log('no support for 3.3 or earlier ', version);
+    return LoadCalChart33AndEarlier(new DataView(buffer, 8, buffer.byteLength-8));
+  }
+  return LoadCalChart3Current(new DataView(buffer, 8, buffer.byteLength-8));
+}
+
+
+// Importing calchart 3.3 and earlier is not supported
+function LoadCalChart33AndEarlier(buffer: DataView) {
+  throw new Error('No support for CalChart 3.3 or ealier');
+}
+
+// calchart 3.4 is a series of tables, with each one have the format: name_32, size_32, data, end_32 
+function LoadCalChart3Current(buffer: DataView) {
+  let show = new Show();
+  // doing this here till we resolve #39 Don't set a default title on a new show
+  show.title = '';
+  show.stuntSheets = [];
+  const split = SplitDataViewIntoChunks(buffer);
+  for (let block of split) {
+    if (block[0] == "SHOW") {
+      show = ParseCalChart3SHOW(block);
+    }
+  }
+  return show;
+}
+
+function ParseCalChart3SHOW(block: [string, DataView]) {
+  let show = new Show();
+  // doing this here till we resolve #39 Don't set a default title on a new show
+  show.title = '';
+  show.stuntSheets = [];
+  const split = SplitDataViewIntoChunks(block[1]);
+  for (let block of split) {
+    if (block[0] == "SIZE") {
+      ParseCalChart3SHOWSIZE(show, block);
+    }
+    if (block[0] == "LABL") {
+      ParseCalChart3SHOWLABL(show, block);
+    }
+    if (block[0] == "DESC") {
+      ParseCalChart3SHOWDESC(show, block);
+    }
+    if (block[0] == "SHET") {
+      ParseCalChart3SHOWSHET(show, block);
+    }
+  }
+  return show;
+}
+
+function ParseCalChart3SHOWSIZE(show: Show, block: [string, DataView]) {
+  if (block[1].byteLength != 4) {
+    throw new Error('Show Size field is incorrect');
```

Comment:
We can keep the error throwing simple unless we need the `Error` object for some reason
```suggestion
    throw 'Show Size field is incorrect'
```

Reply:
Fixed in next update